nym-vpn-core/crates/nym-vpn-network-config/src/refresh.rs

```diff
@@ -1,47 +1,67 @@
 // Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
 // SPDX-License-Identifier: GPL-3.0-only
 
-use std::{
-    path::{Path, PathBuf},
-    time::Duration,
-};
+use std::{path::PathBuf, time::Duration};
 
 use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 
 use super::discovery::Discovery;
 
-fn refresh_discovery_file(config_dir: &Path, network_name: &str) -> anyhow::Result<()> {
-    if !Discovery::path_is_stale(config_dir, network_name)? {
-        return Ok(());
-    }
-    Discovery::fetch(network_name)?.write_to_file(config_dir)?;
-    Ok(())
-}
-
-// Ideally we only refresh the discovery file when the tunnel is up
-#[allow(unused)]
-pub(crate) async fn start_background_discovery_refresh(
+struct DiscoveryRefresher {
     config_path: PathBuf,
     network_name: String,
     cancel_token: CancellationToken,
-) -> JoinHandle<()> {
-    tokio::spawn(async move {
+}
+
+impl DiscoveryRefresher {
+    fn new(config_path: PathBuf, network_name: String, cancel_token: CancellationToken) -> Self {
+        Self {
+            config_path,
+            network_name,
+            cancel_token,
+        }
+    }
+
+    async fn refresh_discovery_file(&self) -> anyhow::Result<()> {
+        tracing::info!("Checking descovery file age");
+        if !Discovery::path_is_stale(self.config_path.as_path(), &self.network_name)? {
+            return Ok(());
+        }
+        tracing::info!("Refreshing and updating file");
+        Discovery::fetch(&self.network_name)
+            .await?
+            .write_to_file(self.config_path.as_path())?;
+        Ok(())
+    }
+
+    async fn run(self) {
         // Check once an hour
-        let mut interval = tokio::time::interval(Duration::from_secs(60 * 60));
+        let mut interval = tokio::time::interval(Duration::from_secs(60));
```

Comment:
Having this check every 60 sec is fine, but then I think the max file age should be bumped to something larger. We don't need to overwrite this file every minute imho

I think the max age used to be something pretty large, every 24h. But it was changed to 60sec since we wanted to make sure to fetch the network messages when updated.

Currently the startup logic is something like

0. write default config if nothing exists
1. check if file is older than max age
2. update if older than max age

But I'm thinking we should change it to

0. write default config if nothing exists
1. always try update file on startup
2. update in the background periodically